### ISSUE TITLE
Allow setting root dir for winrepo_remotes_ng

### ISSUE THREAD

**Issue by jlucas** [saltstack/salt#60471]
@twangboy I can't see a way to set a root directory within the repository for `winrepo_remotes_ng` like there is with `gitfs_remotes` and `ext_pillar`.

The configuration for `gitfs_remotes` allows "name", "base" (tag/branch), and "root" configuration per repository, e.g.
```
          [{"https://foo:abc123@gitlab.com/mycompany/salt.git":
            [{"name": "salt"},
             {"base": "dev"},
             {"root": "salt"}]}],
```

The `ext_pillar` configuration's git mode, allows you to specify a tag/branch (via prepending the branch name to the repository URL) and also supports specifying a "root" within the repo to draw from (and some other options), e.g.
```
        "ext_pillar":
          [{"git": [{"dev https://foo:abc123@gitlab.com/mycompany/salt.git": [{"root": "salt/pillar"}]}]}],
```

The current configuration for `winrepo_remotes_ng` supports specifying a tag/branch with the same prepending syntax `ext_pillar` uses, but there is no way to specify the root directory within the repository.  The only additional information that can be given is a "name", for disambiguation purposes.

The `gitfs_remotes` configuration seems to be richest in terms of options, so it would be nice if `ext_pillar` and `winrepo_remotes_ng` just followed the same pattern.  However, the most critical thing (for us) is addressing the missing functionality that will allow us to specify a root within the repository for `winrepo_remotes_ng` repos.

At a minimum, conform `winrepo_remotes_ng` configution to `ext_pillar`, e.g...
```
"winrepo_remotes_ng": ["https://github.com/saltstack/salt-winrepo-ng.git",
          {"dev https://foo:abc123@gitlab.com/mycompany/salt.git": [{"root": "salt/win"}]}]
```
...but, preferably, conform it to `gitfs_remotes`, e.g...
```
        "winrepo_remotes_ng":
          ["https://github.com/saltstack/salt-winrepo-ng.git",
           {"https://foo:abc123@gitlab.com/mycompany/salt.git":
            [{"name": "mycompany"},
             {"base": "dev"},
             {"root": "salt/win"}]}],

**Comment by jlucas** [comment]
I finally got into the Salt Slack yesterday and asked about this. 

msmith there suggested a workaround of moving the winrepo to gitfs for and mounting it under salt://win/repo-ng.

I did that, with this configuration...

```
"gitfs_remotes":
  [{"https://foo:abc123@gitlab.com/mycompany/salt.git":
    [{"name": "salt-win"},
     {"base": "dev"},
     {"root": "salt/win/repo-ng"},
     {"mountpoint": "salt://win/repo-ng"}]}],
```

...and it has worked around the issue for the time being.

It would still be nice to see all the various git methods have the same configuration options, unless gitfs is meant to replace the other ones.
